zkvm/src/contract.rs

```diff
@@ -91,9 +78,45 @@ impl Input {
             txid,
         })
     }
+
+    // Creates a FrozenContract from payload (which is a vector of (qty, flv)) and pred.
+    // Serializes the contract, and uses the serialized contract and txid to generate a utxo.
+    // Returns an Input with the contract, txid, and utxo.
+    pub fn new(payload: Vec<(Commitment, Commitment)>, pred: Predicate, txid: TxID) -> Self {
+        let frozen_values: Vec<FrozenItem> = payload
```

Comment:
if you shadow `payload` variable, the next line will be shorter...

```suggestion
        let payload: Vec<FrozenItem> = payload
```